src/components/pages/lessons/LessonInfo.tsx

```diff
@@ -153,6 +216,7 @@ const LessonInfo: FunctionComponent<LessonInfoProps> = ({
   lesson,
   ...restProps
 }) => {
+  const hasCode = !Object.values(lesson.code || {}).every((x) => x === null)
```

Comment:
Here I check if `lesson.code` fields values are not `null` (Lodash's `isEmpty` is not useful here)

Reply:
@theianjones I think the graphql endpoint could probably make this **way** simpler by just returning a `code_url` instead of forcing all this hoop jumping into the client.

Reply:
suggestion: @DrShpongle i think we should fix the api because all we need is a url and all of this work is messy and not the UI's business

what do you think?

Reply:
I totally agree, just tried to make best with what I have. But yes - getting just the url will make things cleaner.

Reply:
I added a `code_url` field and a `repo_url` field to the lesson graphql type here https://github.com/eggheadio/egghead-rails/pull/4175/

Reply:
refactored due to the latest changes in API